### ISSUE TITLE
KeyError: 'ENCODING'

### ISSUE THREAD

**Issue by jelmer** [eventable/vobject#40]
Recent changes mean some vcf files now fail to parse

```
[Sun Sep 04 17:15:29.244163 2016] [wsgi:error] [pid 24799] Traceback (most recent call last):
[Sun Sep 04 17:15:29.244206 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/calypso/calypso/webdav.py", line 270, in insert_file
[Sun Sep 04 17:15:29.244245 2016] [wsgi:error] [pid 24799]     item = self.read_file(path)
[Sun Sep 04 17:15:29.244283 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/calypso/calypso/webdav.py", line 265, in read_file
[Sun Sep 04 17:15:29.244335 2016] [wsgi:error] [pid 24799]     item = Item(text, None, path, self.urlpath)
[Sun Sep 04 17:15:29.244375 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/calypso/calypso/webdav.py", line 80, in **init**
[Sun Sep 04 17:15:29.244412 2016] [wsgi:error] [pid 24799]     self.object = vobject.readOne(text)
[Sun Sep 04 17:15:29.244451 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/vobject/vobject/base.py", line 1130, in readOne
[Sun Sep 04 17:15:29.244494 2016] [wsgi:error] [pid 24799]     allowQP))
[Sun Sep 04 17:15:29.244523 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/vobject/vobject/base.py", line 1074, in readComponents
[Sun Sep 04 17:15:29.244548 2016] [wsgi:error] [pid 24799]     vline = textLineToContentLine(line, n)
[Sun Sep 04 17:15:29.244571 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/vobject/vobject/base.py", line 913, in textLineToContentLine
[Sun Sep 04 17:15:29.244595 2016] [wsgi:error] [pid 24799]     'lineNumber' : n})
[Sun Sep 04 17:15:29.244619 2016] [wsgi:error] [pid 24799]   File "/home/jelmer/vobject/vobject/base.py", line 337, in **init**
[Sun Sep 04 17:15:29.244643 2016] [wsgi:error] [pid 24799]     self.value = codecs.decode(self.value.encode("utf-8"), "quoted-printable").decode(self.params['ENCODING'])
KeyError: 'ENCODING'
```

This makes sense when reading the code. In base.py:330, if 'ENCODING' is set to just 'QUOTED-PRINTABLE', that item is first removed from the list and then 'ENCODING' itself is removed because it is empty.

**Comment by jelmer** [comment]
This seems to be a regression from 53fb3178be1b7359844b07b50a6f5ad4577af1ea

**Comment by jelmer** [comment]
And similarly, if 'QUOTED-PRINTABLE' as a singletonparam is set, 'ENCODING' can not be set at all.

**Comment by jelmer** [comment]
I think this is also fixed after #41 was merged?

**Comment by wpercy** [comment]
Yep just want to run some tests

**Comment by wpercy** [comment]
resolved by #49
